nemo/collections/nlp/models/dialogue/intent_bio_type_classification_model.py

```diff
@@ -0,0 +1,821 @@
+        # Update configuration with new data.
+        self._set_data_desc_to_cfg(self.cfg, data_dir, train_ds, validation_ds)
+        # Reconfigure the classifier for different settings (number of intents, slots etc.).
+        self._reconfigure_classifier()
+
+    def update_data_dir_for_testing(self, data_dir) -> None:
+        """
+        Update data directory.
+
+        Args:
+            data_dir: path to data directory
+        """
+        logging.info(f'Setting data_dir to {data_dir}.')
+        self.data_dir = data_dir
+    
+    @staticmethod
+    def get_entity_embedding_from_hidden_states(mention_mask, hidden_states):
+        '''
+        Generate mean pooling embedding per entity mention from hidden_states
+        Args:
+            mention_mask: tensor of size (batch_size, num_tokens, num_entities), that maps tokens to entity matches using BIO slot class of each word token
+            hidden_states: tensor of size (batch_size, num_tokens, hidden_state_dim)
+
+            eg,
+            send me a  wake up alert at seven am tomorrow morning
+            0    0  0  1    2  0     0  1     2  1        1
+            0: 'O' other
+            1: 'B' Begin
+            2: 'I' Inside
+            mention_mask = [[1, 0, 0, 1, 2, 0, 1]]
+            hidden_states = torch.ones((1,7,3))
+            [1, 1, 1], [1, 1, 1], [1, 1, 1], [0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0]
+
+
+        '''
+        _, max_token_len, hidden_states_dim = hidden_states.size()
+        mention_hidden_states=[]
+
+        for one_mention_mask, one_hidden_states in zip(mention_mask, hidden_states):
+            mention_states = []
+            for idx, one_mask_value in enumerate(one_mention_mask):
+                if one_mask_value == 1:
+                    mention_states.append([])
+                if one_mask_value in [1, 2]:
+                    mention_states[-1].append(one_hidden_states[idx])
+            
+            mention_states = [torch.mean(torch.stack(mention_state),0) for mention_state in mention_states]
+            mention_states += [torch.zeros((hidden_states_dim)).to(hidden_states.device) for i in range(max_token_len-len(mention_states))]
+            mention_hidden_states.append(torch.stack(mention_states))
+        
+        return torch.stack(mention_hidden_states)
+
+    @typecheck()
+    def forward(self, input_ids, attention_mask, token_type_ids, mention_mask):
+        """
+        No special modification required for Lightning, define it as you normally would
+        in the `nn.Module` in vanilla PyTorch.
+        """
+        if self._cfg.tokenizer.get('library', '') == 'megatron':
+            hidden_states, _ = self.bert_model(input_ids, attention_mask, tokentype_ids=token_type_ids, lm_labels=None)
+        else:
+            hidden_states = self.bert_model(
+                input_ids=input_ids, token_type_ids=token_type_ids, attention_mask=attention_mask
+            )
+
+        from os.path import exists
```

Comment:
Please remove these global variables / inline import after the code is finalized

Reply:
Will remove later when the code is finalized